tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2035,6 +2031,34 @@ def test_run_query_with_on_demand_count(self):
             ],
         )
 
+    # Once we delete the current spec version this test will fail and we can delete it
+    def test_on_demand_builder_with_new_spec(self):
+        field = "count()"
+        query = "transaction.duration:>0"
+        spec = OnDemandMetricSpec(field=field, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY)
+        # As expected, it does not include the environment tag
+        expected_str_hash = "None;{'name': 'event.duration', 'op': 'gt', 'value': 0.0}"
+        assert spec._query_str_for_hash == expected_str_hash
```

Comment:
Inside of the with block you will see that the query hash will include `["environment"]`.